draft-rokui-ccamp-actn-wdm-pluggable-modelling.md

```diff
@@ -606,270 +564,525 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+Prior to progressing further it is important to note that pluggables are highly valuable, and correspondingly expensive. They are deployed in a controlled fashion. There are a range of policies for deployment of pluggables.
 
-~~~~
+In some cases, at one extreme end of the range of policy choices, an operator may decide to fully populate a packet device with a selection of pluggables and may cable them to adjacent ROADMS. However, it is more likely that pluggable deployment will be on a just-in-time basis, at the other end of the range of policy choices, so a pluggable is not deployed (and hence is not cabled) until the solution to realization of the optical service has been determined.
 
-// To convert the "Coherent Pluggable Google Sheet" to
-// "Coherent Pluggable Yang Data Model", apply the following
-// methodology to all attributes in "Coherent Pluggable Google Sheet"
-
-// Note: This example uses attribute "channel-output-power" just
-//       as an example.
-//       This methodology provided is applicable to any attributes.
-
-    channel-output-power{
-        is-capability-attributes = (TRUE or FALSE)
-        {
-          // if "TRUE", the following needed to be added
-          ro min-channel-output-power = ...
-          ro max-channel-output-power = ...
-          ro default-channel-output-power = ...
-        }
-
-        is-configurable = (TRUE or FALSE)
-        {
-          // if "TRUE", this is R/W attribute
-          rw channel-output-power = ...
-        }
-
-        is-monitorable = (TRUE or FALSE)
-        {
-          // if "TRUE", following PM data from pluggable might
-          // be available and can be read
-          ro current-channel-output-power (if applicable)
-          ro average-channel-output-power (if applicable)
-          ro min-channel-output-power     (if applicable)
-          ro max-channel-output-power     (if applicable)
-        }
-
-        is-threshold-available = (TRUE or FALSE)
-        {
-          // if "TRUE", following PM thresholds might be configurable
-          rw threshold-for-warning-alert  (if applicable)
-          rw threshold-for-minor-alert    (if applicable)
-          rw threshold-for-major-alert    (if applicable)
-          rw threshold-for-critical-alert (if applicable
-        }
-    }
+Regardless of the pluggable deployment approach, the pluggable capability will be accounted for in the optical network analysis activity. Where pluggables are present, the range of installed pluggables constrain the possible realizations, where the pluggables have not been deployed all approved pluggables (type&version) could be considered during the analysis, although capability of the relevant packet devices to support specific pluggables will also need to be considered, and this may eliminate some pluggables. In addition, if there is some urgency, the availability of the type of pluggable to the installation engineer and/or in the local spares holding inventory may also be considered.
```

Comment:
```
Regardless of the deployment approach, the module capability will be accounted for in the optical network analysis activity. Where modules are present, the range of installed modules constrain the possible realizations, where pluggable modules have not been deployed all approved pluggable modules (type&version) could be considered during the analysis, although capability of the relevant packet devices to support specific pluggable modules will also need to be considered, and this may eliminate some pluggable modules. In addition, if there is some urgency, the availability of the type of pluggable modules to the installation engineer and/or in the local spares holding inventory may also be considered.
```